docs/driver.rst

```diff
@@ -0,0 +1,311 @@
+    title: "Iron Man"
+  });
+  console.log(result);
+  // {id: "047c5893..."}
+
+Remember that :ref:`parameters <ref_eql_params>` can only be *scalars* or *arrays of scalars*.
+
+Checking connection status
+--------------------------
+
+The client maintains a dynamically sized *pool* of connections under the hood. These connections are initialized *lazily*, so no connection will be established until the first time you execute a query.
+
+If you want to explicitly ensure that the client is connected without running a query, use the ``.ensureConnected()`` method.
+
+.. code-block:: js
+
+  const edgedb = require("edgedb");
+
+  const client = edgedb.createClient();
+
+  async function main() {
+    await client.ensureConnected();
+  }
+
+.. _edgedb-js-api-transaction:
+
+Transactions
+------------
+
+The most robust way to execute transactional code is to use
+the ``transaction()`` API:
+
+.. code-block:: js
+
+    await client.transaction(tx => {
+      await tx.execute("insert User {name := 'Don'}");
+    });
+
+Note that we execute queries on the ``tx`` object in the above
+example, rather than on the original ``client`` object.
+
+The ``transaction()`` API guarantees that:
+
+1. Transactions are executed atomically;
+2. If a transaction fails due to retryable error (like
+   a network failure or a concurrent update error), the transaction
+   would be retried;
+3. If any other, non-retryable error occurs, the transaction is rolled
+   back and the ``transaction()`` block throws.
+
+The key implication of retrying transactions is that the entire
+nested code block can be re-run, including any non-querying
+JavaScript code. Here is an example:
+
+.. code-block:: js
+
+    const email = "timmy@edgedb.com"
+
+    await client.transaction(async tx => {
+      await tx.execute(
+        `insert User { email := <str>$email }`,
+        { email },
+      )
+
+      await sendWelcomeEmail(email);
+
+      await tx.execute(
+        `insert LoginHistory {
+          user := (select User filter .email = <str>$email),
+          timestamp := datetime_current()
+        }`,
+        { email },
+      )
+    })
+
+In the above example, the welcome email may be sent multiple times if the
+transaction block is retried. Generally, the code inside the transaction block
+shouldn't have side effects or run for a significant amount of time.
+
+.. note::
+
+  Transactions allocate expensive server resources and having
+  too many concurrently running long-running transactions will
+  negatively impact the performance of the DB server.
+
+.. note::
+
+  * RFC1004_
+  * :js:meth:`Client.transaction\<T\>`
+
+  .. _RFC1004: https://github.com/edgedb/rfcs/blob/master/text/1004-transactions-api.rst
+
+
+Next up
+-------
+
+If you're using plain JavaScript, the driver API will likely meet your needs. If you're a TypeScript user and want autocompletion and type inference, head over to the :ref:`Query Builder docs <edgedb-js-qb>`.
```

Comment:
I think the query builder is useful even though typescript is not used. VSCode can give autocomplete suggestions even with js and using functions is better than edgeql query strings for decent usage

Reply:
Updated to reflect this